src/coap_debug.c

```diff
@@ -237,23 +252,30 @@ coap_print_addr(const struct coap_address_t *addr, unsigned char *buf, size_t le
 #  else /* WITH_UIP6 */
 #   warning "IPv4 network addresses will not be included in debug output"
 
-  if (len < 21)
+  if (len < 21) {
+    *p = '\000';
     return 0;
+  }
 #  endif /* WITH_UIP6 */
-  if (buf + len - p < 6)
-    return 0;
+  if (buf + len - p < 6) {
+    *p = '\000';
+    return p - buf;
+  }
 
 #ifdef HAVE_SNPRINTF
-  p += snprintf((char *)p, buf + len - p + 1, ":%d", uip_htons(addr->port));
+  /* Cannot rely on snprintf() return value for short buffers */
+  snprintf((char *)p, buf + len - p - 1, ":%d", uip_htons(addr->port));
```

Comment:
I think the length here should be `buf + len - p` as well.

Reply:
Agreed - fixed